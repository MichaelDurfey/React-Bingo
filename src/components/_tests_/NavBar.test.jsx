import React from 'react';
import renderer from 'react-test-renderer';
import NavBar from '../NavBar';

describe('Message', () => {
  test('it renders correctly with defaults', () => {
    const navbar = renderer
      .create(<NavBar />)
      .toJSON();
    expect(navbar).toMatchSnapshot();
  });
  test('Navbar correctly renders children', () => {
    const navbar = renderer
      .create(<NavBar lastBall={25} played={[1, 2, 3, 4]} />)
      .toJSON();
    expect(navbar.children).toHaveLength(3);
  });
  test('PlayedNumbers div has correct length when passed 4 numbers', () => {
    const navbar = renderer
      .create(<NavBar lastBall={25} played={[1, 2, 3, 4]} />)
      .toJSON();
    expect(navbar.children[2].children[1].children.length).toBe(4);
  });
  test('lastBall exists when passed down', () => {
    const navbar = renderer
      .create(<NavBar lastBall={25} played={[1, 2, 3, 4]} />)
      .toJSON();
    expect(navbar.children[1].children.includes('25')).toBe(true);
  });
});