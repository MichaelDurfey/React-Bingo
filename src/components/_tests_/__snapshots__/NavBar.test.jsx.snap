// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message it renders correctly with defaults 1`] = `
<div
  className="navBar"
>
  <div
    className="lastBallText"
  >
    Last Ball
  </div>
  <div
    className="playedContainer"
  >
    <div
      className="playedText"
    >
      previous balls
    </div>
    <div
      className="playedNumbers"
    />
  </div>
</div>
`;
